src/platform/APM32/platform_mcu.h

```diff
@@ -147,3 +147,28 @@
 
 #define USE_TIMER_MGMT
 #define USE_TIMER_AF
+
+#if defined(APM32F4)
+
+//speed is packed inside modebits 5 and 2,
+#define IO_CONFIG(mode, speed, pupd) ((mode) | ((speed) << 2) | ((pupd) << 5))
```

Comment:
This could easily lead to similar problem as in UART
- only one version will be copied when new platform is added, possibly ignoring some important part in other clone
- most developers will fix bugs only on platform they use. Bug is them fixed multiple times
- (and equivalent of SmartAudio may appear, cyclically breaking and fixing different SA implementations)

Single file is ugly, but all version are close together and it is a bit harder to introduce some problems

Reply:
Yes, I think it's important to distinguish a setting / something simple that's not likely to ever be a bug (outside of a particular platform) versus a file with logic etc. 

It's also a case by case as a single file with lots of different lib ifdef include logic may be more complicated and hard to maintain.

So the option for either is now present (the common location for STM and clone derivatives for example).